### DISCUSSION TITLE
opencv[world] seems to not actually build a standalone DLL

### DISCUSSION THREAD

**Discussion by jwezorek** [microsoft/vcpkg#34579] [General]
I'm trying to build a standalone opencv_world4.dll via
`.\vcpkg.exe install opencv[world]:x64-windows`
The build succeeds but the built DLL has dependencies on other DLLs (I can tell with dependency walker), unlike OpenCV's prebuilt binaries. More details [here](https://stackoverflow.com/questions/77319824/why-is-loadlibrary-failing-to-load-an-opencv-world-dll-built-by-vcpkg-but-not-th).

**Comment by Osyotr** [comment]
You need this https://learn.microsoft.com/en-us/vcpkg/users/triplets#per-port-customization

**Comment by jwezorek** [comment]
I need more information to understand how that helps.

Anyway, I'm building OpenCV from source now and when I set the "world" feature in CMake it indeed builds a DLL that has no dependencies, so it seems to me something is wrong with vcpkg's OpenCV package.

**Reply by Osyotr** [reply]
You're building with `x64-windows` triplet which builds everything as dynamic libraries, including all of the opencv dependencies.

Here's a quote from the doc linked above:
> If you need to apply different settings for different libraries, you must make a single custom triplet with that combination of settings.
See [Per-port customization](https://learn.microsoft.com/en-us/vcpkg/users/triplets#per-port-customization) below for how to accomplish this.

The doc also has an examples how to add and use custom triplets. Basically you need to set `VCPKG_LIBRARY_LINKAGE` to static but override it with `dynamic` for opencv. You probably would also need to set `VCPKG_CRT_LINKAGE` to `static` for everything (including opencv) if you don't want to depend on msvc DLLs.
